src/main/java/hudson/plugins/ec2/util/MinimumInstanceChecker.java

```diff
@@ -0,0 +1,43 @@
+package hudson.plugins.ec2.util;
+
+import hudson.plugins.ec2.EC2Cloud;
+import hudson.plugins.ec2.EC2Computer;
+import hudson.plugins.ec2.SlaveTemplate;
+import jenkins.model.Jenkins;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class MinimumInstanceChecker {
```

Comment:
Unless this class is meant for public use, you can add `@Restricted(NoExternalUse.class)` to this (and anything else that's not meant for use outside of the plugin; it helps avoid restricting backward compatibility).

Reply:
I added that. But I didn't do it for the factories, etc, since I figure someone might actually want to extend those.